### DISCUSSION TITLE
Can`t select drive on DiskCostDlg

### DISCUSSION THREAD

**Discussion by FructusS** [oleg-shilo/wixsharp#1634] [Q&A] [answered]
Hello, I cant select drive on DickCostDlg. I use wixsharp 3 custom dialog. 
![image](https://github.com/user-attachments/assets/636f33e0-d37e-4d9a-a1a8-cb39c659d800)
My program.cs 
```
using System;
using System.Diagnostics;
using System.Windows.Forms;
using WixSharp;
using WixSharp.Controls;
using WixSharp.Forms;
using WixSharpSetup;

namespace WixSharp_Setup1
{
    internal class Program
    {
        static void Main()
        {
            var project = new ManagedProject("MyProduct",
                             new Dir(@"%ProgramFiles%\My Company\My Product",
                                 new File("Program.cs")));

            project.GUID = new Guid("7158421c-9073-4bda-8939-2a2a9cfc6109");

            project.CustomUI = BuildUI();
            project.UI = WUI.WixUI_InstallDir;

            project.OutDir = "C:\\Users\\PC\\Desktop\\tz";

            //ValidateAssemblyCompatibility();

            project.BuildMsi();
        }

        private static CustomUI BuildUI()
        {
            return new DialogSequence().On(NativeDialogs.WelcomeDlg, Buttons.Next, new SpawnDialog(NativeDialogs.DiskCostDlg));
        }
    }
}
```

**Comment by oleg-shilo** [comment, marked as answer]
I do not have the answer so I have marked your post with "please help to answer".

Please be aware that the customization via native WiX dialogs has been deprecated as rather troublesome so WixSharp resorted to the industry-strength UI frameworks WinForms and WPF where you can rely on the whole power of these environments (e.g. your C# and XAML sources).

However, DiskCostDlg has been dropped in the default VS project templates because of a low demand for its use-case. 
But of course, you can still implement it by yourself if you indeed need it.
